### PR TITLE
[transit] Optimization. Creating world graph once for building mwm transit section.

### PR DESCRIPTION
**Pull request by bykoianko**
Пару недель назад, при написании кода поиска ближайшего (лучшего) пешеходного сегмента к гейту мы спрятали создание WorldGraph в метод: IndexRouter::FindBestSegmentInSingleMwm(). Это оказалось не эффективно, поскольку для каждого Gate-та при создании секции transit создавался WorldGraph. Я передал код так, чтоб WorldGraph создавался один раз, на сборку секции transit.

@tatiana-kondakova @ygorshenin PTAL

### CODE REVIEW

File: routing/index_router.cpp [hunk 1]
```diff
@@ -318,13 +318,19 @@ IndexRouter::IndexRouter(VehicleType vehicleType, bool loadAltitudes,
   CHECK(m_directionsEngine, ());
 }
 
-bool IndexRouter::FindBestSegmentInSingleMwm(m2::PointD const & point, m2::PointD const & direction,
-                                             bool isOutgoing, Segment & bestSegment)
+std::unique_ptr<WorldGraph> IndexRouter::MakeSingleMwmWorldGraphForGenerator()
```
Comment (ygorshenin) [inline routing/index_router.cpp]: Maybe it's worth to remove ```ForGenerator``` suffixes?

Comment (bykoianko) [reply routing/index_router.cpp]: Maybe. But in that case we show explicitly that we uncover WorldGraph for generator only. Beside that we write suffix `ForTesting`. Do you think it's better to remove `ForGenerator` anyway?

Comment (ygorshenin) [reply routing/index_router.cpp]: ForTesting is used for methods that expose somehow internal structure of the class, or methods that are doing something that is very dangerous in prod, but ok in tests.
 
These methods looks good to me, not sure that it's worth to specify that they must be used in generator only. Up to you.

Comment (tatiana-kondakova) [inline routing/index_router.cpp]: s/std::unique_ptr/unique_ptr

Comment (bykoianko) [reply routing/index_router.cpp]: Let's remove the suffix.

**Pull request merged**